packages/calcite-components/src/components/input-time-picker/input-time-picker.tsx

```diff
@@ -636,28 +629,40 @@ export class InputTimePicker
   }
 
   private delocalizeTimeStringToParts(
-    localizedTimeString: string,
+    value: string,
     fractionalSecondFormatToken?: "S" | "SS" | "SSS",
   ): DayjsTimeParts {
-    const ltsFormatString = this.localeConfig?.formats?.LTS;
-    const fractionalSecondTokenMatch = ltsFormatString.match(/ss\.*(S+)/g);
-
-    if (fractionalSecondFormatToken && this.shouldIncludeFractionalSeconds()) {
-      const secondFormatToken = `ss.${fractionalSecondFormatToken}`;
-      this.localeConfig.formats.LTS = fractionalSecondTokenMatch
-        ? ltsFormatString.replace(fractionalSecondTokenMatch[0], secondFormatToken)
-        : ltsFormatString.replace("ss", secondFormatToken);
-    } else if (fractionalSecondTokenMatch) {
-      this.localeConfig.formats.LTS = ltsFormatString.replace(fractionalSecondTokenMatch[0], "ss");
+    const effectiveLocale = this.messages._lang;
+    let localizedTimeString = value;
+    const effectiveHourFormat = isLocaleHourFormatOpposite(
+      this.effectiveHourFormat,
+      effectiveLocale,
+    )
+      ? getLocaleOppositeHourFormat(effectiveLocale)
+      : getLocaleHourFormat(effectiveLocale);
+
+    if (localizedTwentyFourHourMeridiems.has(effectiveLocale) && effectiveHourFormat === "12") {
+      const localizedAM = localizedTwentyFourHourMeridiems.get(effectiveLocale).am;
+      const localizedPM = localizedTwentyFourHourMeridiems.get(effectiveLocale).pm;
+      const localizedAMRegEx = new RegExp(String.raw`${localizedAM}`, "g");
+      const localizedPMRegEx = new RegExp(String.raw`${localizedPM}`, "g");
+      const meridiemFormatToken = getMeridiemFormatToken(effectiveLocale);
+      const caseAdjustedAMString =
+        meridiemFormatToken === meridiemFormatToken.toUpperCase() ? "AM" : "am";
+      const caseAdjustedPMString =
+        meridiemFormatToken === meridiemFormatToken.toUpperCase() ? "PM" : "pm";
+
+      localizedTimeString = localizedTimeString.match(localizedPMRegEx)
+        ? localizedTimeString.replaceAll(localizedPMRegEx, caseAdjustedPMString)
```

Comment:
Are the `RegExp`s needed? Since the pattern uses the meridiem string directly and `replaceAll` already matches globally, it might not be necessary.